prediction_market_agent/agents/microchain_agent/nft_treasury_game/scripts/generate_report.py

```diff
@@ -95,11 +101,18 @@ def generate_report(rpc_url: str) -> None:
         # How many NFTs the agents ended the game with.
         nft_balance = get_nft_balance_retry(owner_address=agent_address, web3=w3)
         logger.info(f"{agent_id} {diff_xdai_balance=:.2f} {nft_balance=}")
+        balances_diff.append(
+            {
+                "agent_id": agent_id.value,
+                "xdai_difference": f"{diff_xdai_balance:.2f}",
+                "nft_balance_end": nft_balance,
+            }
+        )
 
     learnings = summarize_prompts_from_all_agents()
-    with open("report.md", "w") as file:
-        file.write(learnings)
-
-
-if __name__ == "__main__":
-    typer.run(generate_report)
+    current_utc_datetime = datetime.utcnow().strftime("%Y-%m-%d-%H-%M-%S")
+    with open(output_dir.joinpath(f"report_{current_utc_datetime}.md"), "w") as file:
```

Comment:
```suggestion
    with open(output_dir / f"report_{current_utc_datetime}.md", "w") as file:
```

In case you don't know this trick